samlauthenticator/samlauthenticator.py

```diff
@@ -659,19 +690,12 @@ def _authenticate(self, handler, data):
             self.log.debug('Authenticated user using SAML')
             username = self._get_username_from_saml_doc(signed_xml, saml_doc_etree)
             username = self.normalize_username(username)
-            user_roles = self._get_roles_from_saml_doc(signed_xml, saml_doc_etree)
-
-            user_roles_result = self._check_role(user_roles)
-            if not user_roles_result:
-                self.log.error('User role not authorized')
-                return None
 
-            self.log.debug('Optionally create and return user: ' + username)
-            username_add_result = self._check_username_and_add_user(username)
-            if username_add_result:
-                return username
+            if self._valid_config_and_roles(signed_xml, saml_doc_etree):
+                self.log.debug('Optionally create and return user: ' + username)
+                return self._check_username_and_add_user(username)
```

Comment:
I like this changes to this function. Its easier to see the checks performed to authenticate a user.